rosbag2_cpp/src/rosbag2_cpp/readers/sequential_reader.cpp

```diff
@@ -27,6 +27,32 @@ namespace rosbag2_cpp
 {
 namespace readers
 {
+namespace details
+{
+std::vector<std::string> resolve_relative_paths(
+  const std::string & base_folder, std::vector<std::string> relative_files)
+{
+  // TODO(Karsten1987): It's extremely weird that relative files are prefixed with a folder name
+  auto base_path = rcpputils::fs::path(base_folder).parent_path();
```

Comment:
There is something extremely weird that the relativ files are being prefixed with the actual folder name:

```
 ➭ ros2 bag info rosbag2_2020_04_01-10_14_58/
Files: rosbag2_2020_04_01-10_14_58/rosbag2_2020_04_01-10_14_58_0.db3
```

I would actually love to change this as it does not correspond to the actual folder structure. However, I could see this as an annoying breakage where rosbags recorded from a previous distribution can't easily be replayed without modifying the metadata.yaml.

We could then either come up with some sort of deprecation warning, letting the user know to remove the leading folder name manually in the metadata.yaml or have some sort of `#ifdef <= ELOQUENT`

Reply:
Could you switch this logic based on metadata version?